src/server.py

```diff
@@ -0,0 +1,32 @@
+"""Something server-y."""
+
+import socket
+
+
+def server():
+    """Start a server at localhost:9999."""
+    server = socket.socket( socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
+    address = ('127.0.0.1', 5001)
+    server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    server.bind(address)
+
+    server.listen(1)
+
+    while True:
+
+        conn, addr = server.accept()
+
+        try:
+            message = ""
+            buffer_length = 10
+            while message[-3:] != 'EOF':
+                part = conn.recv(buffer_length)
+                message += part.decode('utf8')
```

Comment:
So you're decoding each chunk as you get it. Since unicode characters can take up multiple bytes, what happens when a multiple-byte unicode character isn't received all the way in one chunk? i.e. the buffer length cuts it in half?

Rather than decoding one chunk at a time, what would be a better approach?